mkcodes.py

```diff
@@ -79,43 +113,55 @@ def get_files(inputs):
         elif path.suffix in markdown_extensions:
             yield path, path.parent
 
-def add_inits_to_dir(path):
+
+def add_inits_along_path(from_path, to_path):
     """Recursively add __init__.py files to a directory
-    This compensates for https://bugs.python.org/issue23882 and https://bugs.python.org/issue35617
+    This compensates for https://bugs.python.org/issue23882
+    and https://bugs.python.org/issue35617
     """
-    for child in path.rglob('*'):
-        if child.is_dir():
-            (child / '__init__.py').touch()
-
+    to_path = to_path.expanduser().resolve()
+    from_path = from_path.expanduser().resolve()
+    (to_path / '__init__.py').touch()
+    if to_path == from_path:
+        return
+    if to_path.relative_to(from_path):
+        if from_path != to_path.parent:
```

Comment:
I would have thought this line would prevent putting `__init__.py` in the `outputbasedir`, but that doesn't seem to be the case. What is it's purpose?

Reply:
That is the purpose of it, but I should have added a test to verify it!
I'm having trouble getting the code that is in this pull request to rebase on, but I did this:

tests/test.py:
```py
def test_other_languages(self):
    # previous assertions
    self.assertFalse(self._output_path_exists('__init__.py'))
```

mkcodes.py:
```py
def add_inits_along_path(from_path, to_path):
    to_path = to_path.expanduser().resolve()
    from_path = from_path.expanduser().resolve()
    if to_path == from_path:
        return
    (to_path / '__init__.py').touch()
    if to_path.relative_to(from_path):
        add_inits_along_path(from_path, to_path.parent)
```

the new test fails on the old version of add_inits_along_path (verifying your point) and succeeds with the new version of the function.